app/models/mixins/container_archived_mixin.rb

```diff
@@ -0,0 +1,14 @@
+module ContainerArchivedMixin
+  def archived?
+    ems_id.nil?
+  end
+
+  # Needed for metrics
+  def my_zone
+    if ext_management_system.present?
+      ext_management_system.my_zone
+    elsif old_ext_management_system.present?
```

Comment:
Ok. Could you add a comment that for archived records, use old ems zone. This is making an "association" more complex. Comment will help in future when we need to performance fix this lookup.